extensions/agent/src/data/notebookData.ts

```diff
@@ -0,0 +1,260 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as nls from 'vscode-nls';
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+import { AgentUtils } from '../agentUtils';
+import { IAgentDialogData, AgentDialogMode } from '../interfaces';
+import { NotebookDialogOptions } from '../dialogs/notebookDialog';
+import { createConnection } from 'net';
+
+const localize = nls.loadMessageBundle();
+
+export class NotebookData implements IAgentDialogData {
+
+	private readonly NotebookCompletionActionCondition_Always: string = localize('notebookData.whenJobCompletes', 'When the notebook completes');
+	private readonly NotebookCompletionActionCondition_OnFailure: string = localize('notebookData.whenJobFails', 'When the notebook fails');
+	private readonly NotebookCompletionActionCondition_OnSuccess: string = localize('notebookData.whenJobSucceeds', 'When the notebook succeeds');
+
+	// Error Messages
+	private readonly CreateNotebookErrorMessage_NameIsEmpty = localize('notebookData.jobNameRequired', 'Notebook name must be provided');
```

Comment:
If theres no reason to have these as fields, I would just have these be declared const outside the class.